pkg/ruler/rule_query_consistency.go

```diff
@@ -118,3 +118,17 @@ func isQueryingAlertsForStateMetric(metricName string, matchers ...*labels.Match
 
 	return false
 }
+
+// forceStrongReadConsistencyIfQueryingAlertsForStateMetric enforces strong read consistency if from matchers we
+// detect that the query is querying the ALERTS_FOR_STATE metric, otherwise we don't inject any specific read
+// consistency level and we fallback to the tenant's default.
```

Comment:
should the falling back happen in this function or the caller is responsible for it?

Reply:
I think none of them. The fallback means "do not explicitly set the read consistency" which is what this function does.

Reply:
Falling back to the default happens in the ingester, not in the ruler. Maybe i'm nitpicking too much for a comment.

Reply:
I see what you mean now. Let me rephrase the comment.

Reply:
I rephrased it here: https://github.com/grafana/mimir/pull/7193/commits/61c69178714a28bfa431de0ec2798744abddc558